src/averageAges.js

```diff
@@ -14,12 +14,15 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter((person) => (person.sex === 'm' && century)
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+    : person.sex === 'm');
+
+  return men.reduce(
+    (sumAvg, person) => avaregeAge(person.died, person.born, men.length, sumAvg)
+    , 0);
```

Comment:
The logic with the `reduce` method is repeated in each function. Add it to the your `averageAge` function

Reply:
👌 fixed
I've rebuilt the logic following your recommendations, please, check it